src/mux_client_impl/mod.rs

```diff
@@ -0,0 +1,114 @@
+#![allow(
+    missing_docs,
+    missing_debug_implementations,
+    rustdoc::broken_intra_doc_links,
+    unreachable_pub
+)]
+
+use std::ffi::OsStr;
+use std::path;
+
+use tempfile::TempDir;
+
+use tokio::runtime;
+
+use openssh_mux_client::connection::Connection;
+pub(crate) use openssh_mux_client::connection::{ForwardType, Socket};
+
+use super::Error;
+
+pub(crate) mod builder;
+
+mod fd;
+use fd::*;
+
+mod stdio;
+pub(crate) use stdio::{ChildStderr, ChildStdin, ChildStdout};
+
+use super::Stdio;
+
+mod command;
+pub(crate) use command::Command;
+
+mod child;
+pub(crate) use child::RemoteChild;
+
+#[derive(Debug)]
+pub(crate) struct Session {
```

Comment:
I feel like this should just hold a `Connection` directly, rather than each method having to make one itself.

Reply:
Unfortunately, the ssh multiplex protocol is full of unexpected quirks:
 - one connection can only open one session at a time. **After the session exits, the connection is closed**.
 - ssh multiplex master closes the connection when the remote process is terminated with a signal (similar to how the ssh process exits without any error code when the remote is terminated with a signal).

However, the connection can be reused if it is not used to create a session.

Reply:
That's wild.